homeassistant/components/zwave_js/entity.py

```diff
@@ -188,10 +188,19 @@ def generate_name(
         # Only include non empty additional info
         if additional_info := [item for item in (additional_info or []) if item]:
             name = f"{name} {' '.join(additional_info)}"
-        # append endpoint if > 1
-        if (
-            self.info.primary_value.endpoint is not None
-            and self.info.primary_value.endpoint > 1
+
+        # Only append endpoint to name if there are no equivalent values on a lower
+        # endpoint
+        if self.info.primary_value.endpoint is not None and any(
```

Comment:
I'd cache the `primary_value` attribute in a local variable at the top of this method as we use that attribute multiple times here.